apps/test/netsim/NetSimVisualization.js

```diff
@@ -84,15 +89,74 @@ describe("NetSimVisualization", function () {
   var getVizNodeByEntityID_ = function (_type, id) {
     return this.filter(function(element){
       return element instanceof NetSimVizNode &&
-          element.getCorrespondingEntityID &&
-          element.getCorrespondingEntityID() === id;
+          element.getCorrespondingEntityId &&
+          element.getCorrespondingEntityId() === id;
     })[0];
   };
 
+  function createNode(shard, name) {
+    var newNode;
+    NetSimLocalClientNode.create(shard, name, function (e, n) {
+      newNode = n;
+    });
+    assert(newNode !== undefined, "Failed to create a client.");
+    return newNode;
+  }
+
+  it("does not mismap elements across shard reset", function () {
+    // Fake NetSim
+    var netsim = new NetSim();
+    netSimVis = new NetSimVisualization(makeRootDiv(), netsim.runLoop_);
+
+    // Connect to shard
+    testShard = fakeShard();
+    var myNode = createNode(testShard, 'myNode');
+    netSimVis.setShard(testShard);
+    netSimVis.setLocalNode(myNode);
+
+    // Make a couple more nodes, and make sure the visualization picks them up.
+    var node2 = createNode(testShard, 'node2');
+    var node3 = createNode(testShard, 'node3');
+
+    // Check out the list of viz elements
+    assert.equal(3, netSimVis.elements_.length);
+
+    // Introduce a new shard, with new data
+    var testShard2 = fakeShard();
+    var newNode = createNode(testShard2, 'newNode');
+    netSimVis.setShard(testShard2);
+    netSimVis.setLocalNode(newNode);
+    var newNode2 = createNode(testShard2, 'newNode2');
+
+    // Verify: Nodes on new shard have same IDs, different uuids.
+    assert.equal(myNode.entityID, newNode.entityID);
+    assert.notEqual(myNode.uuid, newNode.uuid);
+    assert.equal(node2.entityID, newNode2.entityID);
+    assert.notEqual(node2.uuid, newNode2.uuid);
+
+    // We now have 4 elements; not 5, because the local node is a straight-up
```

Comment:
I don't understand this sentence. What operation does "local node" refer to? is it setLocalNode?